approaches/bulk-data-loading/README.md

```diff
@@ -0,0 +1,118 @@
+    1. `userId` - on both nodes should match your Vendia Share `userId` (i.e. your email address)
+1. Create the Uni
+    1. `share uni create --config registration.json`
+
+Wait about 5 minutes for the Uni to reach a `Running` state.
+
+## Step 2 - Create an API Key
+You'll next want to create an API Key to use to access the **Retailer** node of your Uni.
+
+This Share CLI command can be used to create an API Key (replace the `<your_uni_name>` value, and feel free to modify the other parameter values as well).
+
+```bash
+share node add-api-key --uni <your_uni_name> --node "Retailer" --name "bulk-load-key" --expiry "2030-01-01"
+```
+
+Make sure to save the API Key value returned, as you'll need it in the next step.
+
+## Step 3 - Create a `.share.env` file
+You'll then want to store the API Key from the previous step and the GraphQL URL for the **Retailer** node in a configuration file, which is used in subsequent steps.
+
+1. Change directories to `src`
+    1. `cd ../src`
+1. Create a `.share.env` file
+    1. `echo -e "GQL_URL=\nGQL_APIKEY=\n" >> .share.env`
+1. Copy the API Key value from the previous step into `.share.env`'s `GQL_APIKEY=` property
+1. Copy the GraphQL URL for the **Retailer** node into `.share.env`'s `GQL_URL=` property
+    1. `share get --uni <your_uni_name>`
+
+## Step 4 - Generate Clients Using Vendia Client SDK
+The [Vendia Share Client SDK](https://www.vendia.net/docs/share/vendia-client-sdk) auto-generates source code based on your Uni's schema.  This makes integration with Vendia Share's GraphQL APIs extremely simple.
+
+1. Pull your Uni's schema 
+    1. `share client:pull`
+1. Install the Vendia Share Client SDK package, which also auto-generates client source code into a `.vendia` directory
+    1. `npm install @vendia/client`
+
+## Step 5 - Generate Data to Load
+With your Uni running, an API Key provisioned, and a `.share.env` file in place, you're ready to create a data set to load into Vendia Share.
+
+The [package.json](src/package.json) file contains a script that can be used to generate data that matches the Uni's schema.  The end result will be an Inventory, in CSV form, stored in a file under the `data` directory of this project.
+
+```bash
+npm run generateData
+```
+
+The values in [Constants.js](src/Constants.js) can be tuned to tune, for example, the `inventoryRecordSize` that will be generated using the command above.
+
+## Step 6 - Execute the Data Loader
+With generated data in place, you're ready to load that data set into Vendia Share.
+
+The [package.json](src/package.json) file also contains a script that can be used to load data that matches the Uni's schema.  The end result will be a Uni populated with the Inventory generated in the previous step.
+
+```bash
+npm run loadData
+```
+
+The values in [Constants.js](src/Constants.js) can be tuned to tune, for example, the `batchSize` and `clientConcurrency` that will be used when loading the data.
```

Comment:
Just to confirm it is intentional to have this comment twice, on this line and line 104.